### ISSUE TITLE
Interleaved_to_sharded hangs for first conv in resnet-50

### ISSUE THREAD

**Issue by TT-BrianLiu** [tenstorrent-metal/tt-metal#4336]
This test is hanging in interleaved_to_sharded (fast and slow dispatch both fails):
```
pytest tests/tt_eager/python_api_testing/unit_testing/test_resnet50_untilize_with_halo_and_conv_v2.py -k LoFi-activations_BFLOAT8_B-weights_BFLOAT8_B-64-16-115-115-4-4-1-1-0-0-batch_8
```

@tarafdarTT  It was passing before with your sharding changes, but after your recent changes and merging, it seems like this is hanging. It's similar to #4319 where unit tests are failing, but resnet-50 is fine since we don't create inputs like this. 

I will put a branch here to reproduce, as well as an older branch that was passing.

**Comment by mywoodstock** [comment]
So it looks like `interleaved_to_sharded` has issues with `ROW_MAJOR` input tensor?

**Comment by TT-BrianLiu** [comment]
Fixed with 07906e318753a2ea3cf119f
